### PR TITLE
Use same dependencies as docker container in alphabetic order

### PR DESCRIPTION
**Pull request by sharkwouter**
There were a couple of dependencies missing and a couple which were not needed in both the readme and the dependency install script. Here is a list:
- Cmake and Doxygen were not in the list at the top.
- The list at the top was almost alphabetically ordered, but not quite.
- The command for installing the dependencies on Ubuntu had build-essential and libsdl-dev in it, which are not required.

Now I do really wonder if Doxygen is used when building the toolchain. The toolchain script does check if it is there, but that was the case for SDL as well, which wasn't needed either.

### CODE REVIEW

**Comment by carstene1ns** [comment]
- `doxygen` is used by some libraries for generating documentation (e.g. opentri)
- `build_essential` is a meta package that provides `gcc`, `g++` and `make`.
These are explicitly included already, but it should be included if the debian folks ever decide to rename packages or something.

**Comment by sharkwouter** [comment]
Ah, that explains the doxygen mystery. I did find it is being used by 2 packages when I build the toolchain again.

Since we were already telling people explicitly to install gcc, g++ and make, we may as well leave out build-essential, in case more which we don't need is added to it in future versions.

**Comment by carstene1ns** [comment]
I would like to merge this after adaption in/from #118.

**Comment by sharkwouter** [comment]
This should be mergable now

**Comment by sharkwouter** [comment]
The following were added after making this merge request and will be removed if this is merged:
- xz-utils
- bzip2
- gzip

I'm not sure if these are really necessary. Especially xz-utils seems odd. gzip and bzip2 are part installed by default on most systems, but setting tar instead may make more sense..

**Comment by sharkwouter** [comment]
Okay, I see now, this was added in commit 5fc6ae5de6194252e4177a4fd0d71804d634b342. Will add them back.

**Comment by carstene1ns** [comment]
gmp, mpfr and mpc should be removed, they are not needed and will be built.
The compression formats should be listed explicitly, as we cannot guarantee archives get extracted after switching to them.

**Comment by sharkwouter** [comment]
I've added them back in and included tar. I'm currently testing to see if the toolchain builds with the dependencies you mentioned removed in from the Dockerfile, will update when it's done building.

**Comment by sharkwouter** [comment]
@carstene1ns you were right, no issues with those dependencies not installed.

**Comment by sharkwouter** [comment]
I've now added pkg-config as well.

File: prepare-debian-ubuntu.sh [hunk 1]
```diff
@@ -1,14 +1,4 @@
...
-  libelf-dev libfreetype6-dev zlib1g-dev libtool libtool-bin subversion git \
-  tcl wget unzip bzip2 gzip xz-utils
-
-# Make `/bin/sh` an alias for `/bin/bash` instead of `/bin/dash` - which is
-# faster, but doesn't play nice with some autotools scripts in psp-ports.
-# The `sudo true` is to make sure we don't pipe into a `sudo` password prompt
-# instead of the intended program.
-sudo true; echo "dash dash/sh boolean false" | sudo debconf-set-selections
-sudo dpkg-reconfigure --frontend=noninteractive dash
+sudo apt-get install $@ autoconf automake bison bzip2 cmake doxygen flex g++ gcc git gzip libelf-dev libgmp-dev libmpfr-dev libncurses5-dev libreadline-dev libtool-bin libusb-dev make mpc patch pkg-config subversion tar tcl texinfo unzip wget xz-utils
```
Comment (carstene1ns) [inline prepare-debian-ubuntu.sh:4]: Can you split the long lines again?

**Comment by carstene1ns** [comment]
What about pacman libs?

File: prepare-debian-ubuntu.sh (hunk 1 above)
Comment (sharkwouter) [reply prepare-debian-ubuntu.sh]: Done!

**Comment by sharkwouter** [comment]
Okay, I've added the pacman dependencies to the README and the debian/ubuntu prepare script.

File: prepare-debian-ubuntu.sh [hunk 2]
```diff
@@ -1,14 +1,9 @@
...
-
-# Make `/bin/sh` an alias for `/bin/bash` instead of `/bin/dash` - which is
-# faster, but doesn't play nice with some autotools scripts in psp-ports.
-# The `sudo true` is to make sure we don't pipe into a `sudo` password prompt
-# instead of the intended program.
-sudo true; echo "dash dash/sh boolean false" | sudo debconf-set-selections
-sudo dpkg-reconfigure --frontend=noninteractive dash
+sudo apt-get install $@ \
+  autoconf automake bison bzip2 cmake doxygen flex g++ gcc git gzip \
+  libarchive-dev libcurl4-openssl-dev libelf-dev libgmp-dev libmpfr-dev \
```
Comment (carstene1ns) [inline prepare-debian-ubuntu.sh:6]: gmp and mpfr

File: prepare-debian-ubuntu.sh [hunk 3]
```diff
@@ -1,14 +1,9 @@
...
+  libncurses5-dev libreadline-dev libssl-dev libtool-bin libusb-dev make mpc \
```
Comment (carstene1ns) [inline prepare-debian-ubuntu.sh:7]: mpc

File: prepare-debian-ubuntu.sh (hunk 2 above)
Comment (sharkwouter) [reply prepare-debian-ubuntu.sh]: Guess I overlooked those, I've removed them.

**Pull request merged**